From 5115c4ce8a635de48c2a2d2e9ce4d5aee74e34bf Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Fri, 20 May 2016 11:50:21 -0700
Subject: [PATCH 0603/1164] HIVE-13783: Display a secondary prompt on beeline
 for multi-line statements (Vihang Karajgaonkar,
 reviewed by Reuben Kuhnert and Sergio Pena)

Change-Id: I96f51c5a1e1a619d72b03275ef1399e2b7778472
---
 .../src/java/org/apache/hive/beeline/Commands.java |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/Commands.java b/beeline/src/java/org/apache/hive/beeline/Commands.java
index 2ecbbe9..2fa2978 100644
--- a/beeline/src/java/org/apache/hive/beeline/Commands.java
+++ b/beeline/src/java/org/apache/hive/beeline/Commands.java
@@ -946,8 +946,8 @@ public String handleMultiLineCmd(String line) throws IOException {
     while (beeLine.getConsoleReader() != null && !(line.trim().endsWith(";")) && beeLine.getOpts()
         .isAllowMultiLineCommand()) {
 
+      StringBuilder prompt = new StringBuilder(beeLine.getPrompt());
       if (!beeLine.getOpts().isSilent()) {
-        StringBuilder prompt = new StringBuilder(beeLine.getPrompt());
         for (int i = 0; i < prompt.length() - 1; i++) {
           if (prompt.charAt(i) != '>') {
             prompt.setCharAt(i, i % 2 == 0 ? '.' : ' ');
@@ -959,7 +959,7 @@ public String handleMultiLineCmd(String line) throws IOException {
       if (beeLine.getOpts().isSilent() && beeLine.getOpts().getScriptFile() != null) {
         extra = beeLine.getConsoleReader().readLine(null, jline.console.ConsoleReader.NULL_MASK);
       } else {
-        extra = beeLine.getConsoleReader().readLine(beeLine.getPrompt());
+        extra = beeLine.getConsoleReader().readLine(prompt.toString());
       }
 
       if (extra == null) { //it happens when using -f and the line of cmds does not end with ;
-- 
1.7.9.5

